mindsdb/interfaces/storage/fs.py

```diff
@@ -314,3 +343,127 @@ def __call__(self, resource_id: int):
             sync=self.sync,
             resource_id=resource_id
         )
+
+
+class ModelStorage:
+    def __init__(self, company_id, predictor_id):
+
+        storageFactory = FileStorageFactory(
+            resource_group=RESOURCE_GROUP.PREDICTOR,
+            company_id=company_id,
+            sync=True
+        )
+
+        self.fileStorage = storageFactory(predictor_id)
+
+        self.company_id = company_id
+        self.predictor_id = predictor_id
+
+    # -- fields --
+
+    def get_info(self):
+        rec = db.Predictor.query.get(self.predictor_id)
+        return dict(status=rec.status, to_predict=rec.to_predict)
+
+    def status_set(self, status, status_info=None):
+        rec = db.Predictor.query.get(self.predictor_id)
+        rec.status = status
+        if status == PREDICTOR_STATUS.ERROR and status_info is not None:
+            rec.data = status_info
+        db.session.commit()
+
+    def columns_get(self):
+        rec = db.Predictor.query.get(self.predictor_id)
+        return rec.dtype_dict
+
+    def columns_set(self, columns):
+        # columns: {name: dtype}
+
+        rec = db.Predictor.query.get(self.predictor_id)
+        rec.dtype_dict = columns
+        db.session.commit()
+
+    # files
+
+    def file_get(self, name):
+        return self.fileStorage.file_get(name)
+
+    def file_set(self, name, content):
+        self.fileStorage.file_set(name, content)
+
+    def file_list(self):
+        ...
+    def file_del(self, name):
+        ...
+
+    # jsons
+
+    def json_set(self, name, data):
+        content = json.dumps(data).encode()
+        self.file_set(f'json_{name}', content)
+
+    def json_get(self, name):
+        content = self.file_get(f'json_{name}').decode()
+        return json.loads(content)
```

Comment:
Please take a look at `mindsdb/interfaces/storage/json.py` - we have json storage in db

Reply:
switched to it